shared/src/test/scala/mlscript/DiffTests.scala

```diff
@@ -52,6 +52,7 @@ class DiffTests extends org.scalatest.funsuite.AnyFunSuite with org.scalatest.Pa
       override def emitDbg(str: String): Unit = output(str)
     }
     var ctx: typer.Ctx = typer.Ctx.init
+    var selfDefineRes: Bool = false
```

Comment:
Better avoid mutable variables whenever possible. Here it should be easy to do away with it!

Reply:
Hmmm... I'm not convinced we should do something different if a `res` was defined somewhere else in the file. Imagine someone adds a `res = 123` at the beginning of a 1000-line fine, and this ends up changing a totally unrelated part of the test file that relied on auto-`res` variable. Better just be consistent here and override any previous `res`.

Reply:
The Yicong test is fine. Note that in principle no `res` is introduced when the results of the lines are already bound.

> I think it is acceptable to override the previous res when the auto-res is updated, and the result of the test above should be like this:

Agreed! Do you see how to do that?

Reply:
Yeah I also noticed `def`s currently seems to modify `res`. But it looks like it may just be a `:js` printing error as `res` is visibly not modified.

> Since res + 1 is not bound to any variable, so I think this should override the res1, and the final result of res + 1 should be 2.

Seems so, yes.

Another thing that might cause problems:

```scala
def res: bool

42

res
```

Reply:
@chengluyu any better ideas?

Reply:
I was mimicking the behavior of Node.js actually. In Node.js REPL, `_` represents the latest result. But if you override with something like `const _ = 8`, you will lost the access to `_` forever.

Reply:
Let's not use JS as a reputable source to take inspiration from :P

Reply:
Yes you should.

Reply:
Thanks I'll take a look when I can (currently tight with time because of a deadline on July 7th).